File: 1-memento/java/memento-v2-datamodel/Originator.java
public class Originator {
    private DataModel data;

    public DataModel getData() {
        return this.data;
    }

    public void setData(DataModel data) {
        this.data = data;
    }

    public void printData() {
        this.data.print();
    }

    public Memento createMemento() {
        return new Memento(this.data);
    }

    public void setMemento(Memento state) {
        this.data = state.getStateData();
    }
}

File: 1-memento/js/implementation-v2-datamodel/Originator.js
const  Memento = require("./Memento.js")
const DataModel = require("./DataModel.js");

module.exports = class Originator {
    #state = new DataModel();
    setState(state) {
        if(!(state instanceof DataModel)) throw new Error("state must be a DataModel");
        this.#state = state;
    }
    getState() {
        return this.#state;
    }
    printState() {
        this.#state.print();
    }
    createMemento() {
        return new Memento(this.#state);
    }
    restoreMemento(memento) {
        if(!(memento instanceof Memento)) throw new Error("memento must be a Memento");
        this.#state = memento.getState();
    }
}